### ISSUE TITLE
Test fail on master

### ISSUE THREAD

**Issue by yancyribbens** [rust-bitcoin/rust-bitcoin#3205]
I'm not sure rn why this isn't being pickedup by CI, however the latest version of master is throwing a test failure.

steps to reproduce:

1) pull the latest version of master branch
2) cp Cargo-minimal.lock Cargo.lock
3) cargo test

**Comment by yancyribbens** [comment]
test output:

```
failures:

---- src/sha256t.rs - sha256t::sha256t_hash_newtype (line 169) stdout ----
error[E0433]: failed to resolve: use of undeclared crate or module `io`
  --> src/sha256t.rs:171:1
   |
5  | / sha256t_hash_newtype! {
6  | |     /// Optional documentation details here.
7  | |     /// Summary is always generated.
8  | |     pub struct FooTag = hash_str("foo");
...  |
13 | |     pub struct FooHash(_);
14 | | }
   | |_^ use of undeclared crate or module `io`
   |
   = note: this error originates in the macro `sha256t_hash_newtype` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of undeclared crate or module `io`
  --> src/sha256t.rs:171:1
   |
5  | / sha256t_hash_newtype! {
6  | |     /// Optional documentation details here.
7  | |     /// Summary is always generated.
8  | |     pub struct FooTag = hash_str("foo");
...  |
13 | |     pub struct FooHash(_);
14 | | }
   | |_^ use of undeclared crate or module `io`
   |
   = note: this error originates in the macro `sha256t_hash_newtype` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0433`.
Couldn't compile the test.

failures:
    src/sha256t.rs - sha256t::sha256t_hash_newtype (line 169)

test result: FAILED. 6 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.42s
```

**Comment by yancyribbens** [comment]
tested with rustc 1.63

**Comment by apoelstra** [comment]
This one is fixed by #3200.

We had a couple bugs in our CI setup, and a couple unrelated bugs in my local CI setup, which led to this slipping through.

**Comment by yancyribbens** [comment]
Ah whops this is not news then. I tested tobins branch and it passed.
